File: SCT2/trunk/test-plugins/org.yakindu.sct.simulation.core.sexec.test/test-gen/org/yakindu/sct/simulation/core/sexec/test/TypeAliasTest.java
/**
 * Copyright (c) 2015 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.simulation.core.sexec.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import org.yakindu.sct.test.models.SCTUnitTestModels;
import com.google.inject.Inject;
import static org.junit.Assert.assertTrue;
/**
 *  Unit TestCase for TypeAlias
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class TypeAliasTest extends AbstractExecutionFlowTest {
	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("TypeAlias.sct");
		initInterpreter(flow);
	}
	@Test
	public void TypeAliasTest() throws Exception {
		interpreter.enter();
		assertTrue(isActive("Start"));
		assertTrue(getInteger("myVar") == 1);
		interpreter.runCycle();
		assertTrue(isActive("Mid"));
		assertTrue(getString("myString").equals("TypeSystem"));
		interpreter.runCycle();
		assertTrue(isActive("Mid2"));
		raiseEvent("myEvent");
		interpreter.runCycle();
		assertTrue(isActive("End"));
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/TypeAliasTest/TypeAliasTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "TypeAlias.h"

TypeAlias handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		typeAlias_init(&handle);
	}
};


TEST_F(StatemachineTest, TypeAliasTest) {					
	typeAlias_enter(&handle);
	EXPECT_TRUE(typeAlias_isStateActive(&handle, TypeAlias_main_region_Start));
	EXPECT_TRUE(typeAliasIface_get_myVar(&handle)== 1l);
	typeAlias_runCycle(&handle);
	EXPECT_TRUE(typeAlias_isStateActive(&handle, TypeAlias_main_region_Mid));
	EXPECT_TRUE(strcmp(typeAliasIface_get_myString(&handle), "TypeSystem") == 0);
	typeAlias_runCycle(&handle);
	EXPECT_TRUE(typeAlias_isStateActive(&handle, TypeAlias_main_region_Mid2));
	typeAliasIface_raise_myEvent(&handle);
	typeAlias_runCycle(&handle);
	EXPECT_TRUE(typeAlias_isStateActive(&handle, TypeAlias_main_region_End));
}


